src/lib/recurringPurposes.js

```diff
@@ -0,0 +1,175 @@
+import log from 'cozy-logger'
+
+import { getManualPurpose, getAutomaticPurpose } from 'src/lib/trips'
+import { setAutomaticPurpose } from 'src/lib/timeseries'
+import {
+  builTimeserieQueryByAccountIdAndStartPlaceAndEndPlace,
+  queryTimeserieByDocId
+} from 'src/queries/queries'
+
+// Similar trips = same start/end dipslay name
+const findSimilarTimeseries = async (client, timeserie) => {
+  const accountId = timeserie.cozyMetadata.sourceAccount
+  const startPlaceDisplayName = timeserie.aggregation.startPlaceDisplayName
+  const endPlaceDisplayName = timeserie.aggregation.endPlaceDisplayName
+  const queryDef = builTimeserieQueryByAccountIdAndStartPlaceAndEndPlace({
+    accountId,
+    startPlaceDisplayName,
+    endPlaceDisplayName
+  }).definition
+  const timeseries = await client.queryAll(queryDef)
+  return timeseries.filter(ts => ts._id !== timeserie._id)
+}
+
+export const findClosestWaybackTrips = async (client, timeserie) => {
+  const waybackTrips = []
+  const accountId = timeserie.cozyMetadata.sourceAccount
+  const startPlaceDisplayName = timeserie.aggregation.endPlaceDisplayName
+  const endPlaceDisplayName = timeserie.aggregation.startPlaceDisplayName
+  // Find closest wayback in the future
+  const queryDefForward = builTimeserieQueryByAccountIdAndStartPlaceAndEndPlace(
+    {
+      accountId,
+      startPlaceDisplayName,
+      endPlaceDisplayName,
+      startDate: timeserie.endDate,
```

Comment:
I find this API confusing. First I was saying "but it will not work if we stay 1h at this place, since the endDate will not be the startOne". Then after reading the `builTimeserieQueryByAccountIdAndStartPlaceAndEndPlace` method, I saw that we use `searchForward` attribute to know what to do `gt / lt`. 

Why not having directly something like: 
```
startDate: {
'$gt': timeserie.endDate
}
```

with no new api to know?

Reply:
By doing this you still have to know about the API and correctly define the startDate predicate. But ok to change this if you find this confusing

Reply:
[c506d38](https://github.com/cozy/coachCO2/pull/174/commits/c506d38ffd7987c967df17ff3b6e2c1733a625de)